### ISSUE TITLE
Support element-wise multiplication

### ISSUE THREAD

**Issue by delashum** [pgvector/pgvector#140]
Saw this is in #27 already but wanted to +1 element-wise multiplication. Currently element-wise addition and subtraction are supported, seems like multiplication and division make sense to add as well.

My use case is that I want to narrow the the vector dimensions that are being compared with the distance operators by multiplying a vector by a standard vector (e.g. [1,0,0,1,0,...]) to zero out some of the values before calculating distance.

**Comment by ankane** [comment]
Hi @delashum, thanks for sharing. Can you give more context on the high level use case?

**Comment by mikeizbicki** [comment]
PR https://github.com/pgvector/pgvector/pull/13 from 2 years ago has working implementations of these routines along with test cases, but was not merged. @delashum  the code in that pr should be a drop in replacement for this repo that does what you want.

Also, quoting <https://github.com/pgvector/pgvector/issues/141#issuecomment-1569416997>:

> @ankane I realize there is legitimate cognitive overhead to including functions that you didn't write in the library, but since these don't affect the "core" functionality of the library at all, I think they'd be relatively easy to merge. There have been a handful of issues now that you've closed because the use cases of this feature are "not common", but perhaps the fact that so many different people have asked for this feature implies that this is actually a commonly sought feature worth the mental overhead on your part.

**Comment by delashum** [comment]
Sure, my use case is that I have a dataset of objects that I want to do some clustering on. The data is partially labelled so I am using some statistical analysis to find which dimensions of the vector indicate similarity stronger than others for the already labelled portion of the data. Then zeroing all other dimensions out in order to cluster new objects only using the similar metrics that are found to be important.

i.e. I want to ignore all dimensions of the vector when calculating distance that haven't been found to indicate similarity in the specific ways I'm looking for.

Currently just doing this in code and seems quite promising, would obviously be great to do as much of this in SQL though so I don't have to pull the entire dataset. Ideally looking to do something like this:
```sql
// weights = [0,0,1,1,1,0,1,0,0,1,...]
SELECT * FROM data WHERE embedding * weights <=> reference_embedding * weights < some_threshold;
```

**Comment by ankane** [comment]
Just fyi, this was added in 08e7209810da8023bd9cdc52e181edc59fc3ccde and will be included in 0.5.0. Thanks again for sharing your use case.
